internal/car/car.go

```diff
@@ -84,14 +88,72 @@ func (c *car) List(ctx context.Context, tag, platform string) error {
 			break
 		}
 	}
-
 	unmatched := pm.Unmatched()
 	if len(unmatched) > 0 {
 		return fmt.Errorf("%s not found in layer", strings.Join(unmatched, ", "))
 	}
 	return nil
 }
 
+func (c *car) Extract(ctx context.Context, tag, platform, directory string, stripComponents int) error {
+	// maintain a lazy map of directories already created
+	dirsCreated := map[string]bool{}
+	return c.do(ctx, func(name string, size int64, mode os.FileMode, modTime time.Time, reader io.Reader) error {
+		destinationPath, ok := newDestinationPath(name, directory, stripComponents)
+		if !ok {
+			return nil // skip
+		}
+
+		baseDir := path.Dir(destinationPath)
+		if ok := dirsCreated[baseDir]; !ok {
+			if err := os.MkdirAll(baseDir, 0755); err != nil { //nolint:gosec
+				return err
+			}
+			dirsCreated[baseDir] = true
+		}
```

Comment:
what if two stripped paths become exactly the same?

Reply:
same as tar I guess, if you have -q first wins else last

Reply:
I'll do a quick test to verify this! whatever it is, we should emulate tar I think

Reply:
it is complicated to answer because overwrite controls are different between bsd tar and gnu tar. However we can say that tar is a stream and you can even have the same file regardless of this flag which changes the name of it.

I can backfill a test about redundant files are overwritten unless `-q` since we haven't yet implemented the myriad of other controls available to gtar for example.

sg?

Reply:
sg! thanks for clarifying